source/agora/consensus/data/EnrollmentData.d

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+
+    Contains supporting code for a enrollment data used in enrollment process.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.data.EnrollmentData;
+
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.Deserializer;
+import agora.common.Hash;
+import agora.common.Serializer;
+import agora.common.crypto.Schnorr : Signature;
+
+/*******************************************************************************
+
+    Define enrollment data (enrollment data = validator)
+
+*******************************************************************************/
+
+public struct EnrollmentData
+{
+    /// K: UTXO key, A public key matching a frozen UTXO
+    public PublicKey K;
```

Comment:
It is correct to set the frozen UTXO hash here.
It is the only way to tell if you have frozen UTXO.

Reply:
Are you asking to change something here?

Reply:
You mean that a enrollment data must have a information like the UTXO hash by which the frozen UTXO validator(you are now implementing) could find a mathcing UTXO.

Reply:
@linked0 @AndrejMitrovic  
Yes, that’s right.
In the Enrollment process, Publickey cannot be a key.
A Publickey can contain many UTXOs.
We have to choose one frozenUTXO.
This can be a key in Enrollment.

Reply:
> We have to choose one frozen UTXO.

True. We could also sum the frozen UTXOs for the given key, but this might complicate the `N` value which is currently hardcoded by definition.

Reply:
Can I apply this request to my next PR? I have already implemented a API endpoint of Node about the enrollment process and I think we need more time to conclude how it can be implemented. So it would be better to merge the code of this PR to the Upstream.

Reply:
Where is the rush? :) If we know the code will be changed after merging we might as well change it before merging. It's very easy to go from the utxo hash to the PublicKey. It's just `utxo_set[utxo_hash].address`.

Reply:
@Geod24 Which do you think is best way to apply @TrustHenry's request between replacing a public key with a UTXO hash or just adding a UTXO hash member in EnrollmentData structure?